model/src/main/kotlin/FileFormat.kt

```diff
@@ -94,9 +94,14 @@ fun File.mapper() = FileFormat.forFile(this).mapper
 fun File.readTree(): JsonNode = mapper().readTree(this)
 
 /**
- * Use the Jackson mapper returned from [File.mapper] to read an object of type [T] from this file.
+ * Use the Jackson mapper returned from [File.mapper] to read a single object of type [T] from this file.
  */
-inline fun <reified T : Any> File.readValue(): T = mapper().readValue(this)
+inline fun <reified T : Any> File.readValue(): T {
+    val mapper = mapper()
+    val parser = mapper.factory.createParser(this)
+    return mapper.readValues<T>(parser).readAll().singleOrNull()
+        ?: throw IllegalArgumentException("Multiple documents found in file '$this' which is not supported.")
```

Comment:
Can this exception be thrown in case the JSON / YAML is valid, but empty?

Reply:
That case throws a Jackson exception derived from `IOException` instead, so I've now aligned on that type.

Reply:
Hm, how would a valid but empty YAML actually look like? ...like the following?
```
----
{}
```

Reply:
I guess just

```
---
```

But that's not a case that we've explicitly handled before, nor do we now.

Reply:
Turns out the above would parse into a `TextNode` with an empty string value.

Reply:
So, could the function still be used for reading e.g. a package curations file which is valid but does not contain any curation? (Or would it now throw?)

Reply:
Please provide and example of the exact (non-0-byte-size) file you have in mind.

Reply:
> Please provide and example of the exact (non-0-byte-size) file you have in mind.

I didn't realize you've edited your top post meanwhile.

Anyway, as those input file seem to be pathological cases, could we agree that handling them (in whatever way) is out of scope of this change?

Reply:
> Anyway, as those input file seem to be pathological cases, could we agree that handling them (in whatever way) is out of scope of this change?

What I'm after is that there should be still a way, to have a `package-curations.yml` file with zero curations in it, which can be successfully read by ORT (and maybe for similar config file cases). (I was not certain whether this PR is a functional change in that regard).

Reply:
I'm confused by the four `----`, shouldn't it be three `---`?

Reply:
Also, I'm confused as you sometimes give examples for `PackageConfiguration`, and sometimes for `PackageCurations`. Let's try to dissect [these](https://github.com/oss-review-toolkit/ort/pull/8741#discussion_r1633136442):

> 1. `readValue<PackageConfiguration>()`

This obviously just works.

> 2. `readValue<PackageConfiguration?>()`

This never worked as `T` is non-nullable for `readValue`.

> 3. `readValue<List<PackageConfiguration>>()`

This *would* work if ORT would allow multiple `PackageConfiguration`s per fiel, but it doesn't (that's what #6873 is all about).

> 4. `readValue<List<PackageConfiguration>?>()`

This again never worked as `T` is non-nullable for `readValue`.

Reply:
I've added two more tests for input with no content.

Reply:
> I'm confused by the four `----`, shouldn't it be three `---`?

This was a typo / Yes, should be three.

> Also, I'm confused as you sometimes give examples for `PackageConfiguration`, and sometimes for `PackageCurations`. Let's try to dissect [these](https://github.com/oss-review-toolkit/ort/pull/8741#discussion_r1633136442):

IIUC, the function is not only called for reading `PackageConfiguration`s, but also for other things, such as `PackageCuration`s. I deliberately mentioned `PackageCuration`s, because it supports a _single_  file containing a list of curations. So, I wanted to ensure that it remains possible to specify a file which contains an empty list of curations, without making throw an exception. The caller is here: [The `FilePackageCuration](https://github.com/oss-review-toolkit/ort/blob/e20d80f0a0c92fb1d1da61fab577fe4a2fbdcd0c/plugins/package-curation-providers/file/src/main/kotlin/FilePackageCurationProvider.kt#L111C34-L111C43)

Reply:
> So, I wanted to ensure that it remains possible to specify a file which contains an empty list of curations, without making ORT throw an exception. The caller is here: [The `FilePackageCuration](https://github.com/oss-review-toolkit/ort/blob/e20d80f0a0c92fb1d1da61fab577fe4a2fbdcd0c/plugins/package-curation-providers/file/src/main/kotlin/FilePackageCurationProvider.kt#L111C34-L111C43)

Right, but don't the [tests added in the last commit](https://github.com/oss-review-toolkit/ort/pull/8741/commits/363fadb7f8af7439c183c8216498ae3b4829c2c4) now cover this?

Reply:
> So, I wanted to ensure that it remains possible to specify a file which contains an empty list of curations, without making ORT throw an exception.

To put it sort, this remains possible, but none of your provided syntax is or ever was valid for an empty list of curations, IMO. Instead, [the syntax as used in the tests](https://github.com/oss-review-toolkit/ort/pull/8741/commits/363fadb7f8af7439c183c8216498ae3b4829c2c4#diff-ecb56d78c099387b7a4c6ef5ac09bbaaad4c1409f7db1ef530bae4b19dfed365R102) needs to be used.